polars/polars-time/src/chunkedarray/datetime.rs

```diff
@@ -41,6 +42,18 @@ pub trait DatetimeMethods: AsDatetime {
         cast_and_apply(self.as_datetime(), temporal::year)
     }
 
+    /// Extract year from underlying NaiveDate representation.
+    /// Returns whether the year is a leap year.
+    fn is_leap_year(&self) -> BooleanChunked {
+        let ca = self.year();
```

Comment:
It's a bit wasteful to first allocate a whole year array. We can determine the year and leap year in one pass while iterating.

Reply:
Good point! I translated it into a kernel, hope that does it. I tried to limit repetitiveness but there was still a bit of code that I needed to write to get this to work.